validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java

```diff
@@ -92,11 +100,44 @@ static ValidatorLoader create(
           metricsSystem,
           validatorSources);
       addLocalValidatorSource(spec, config, slashingProtector, asyncRunner, validatorSources);
+      if (validatorPath.isPresent()) {
+        addMutableValidatorSource(
+            spec, config, slashingProtector, asyncRunner, validatorSources, validatorPath);
+      }
     }
 
     return new ValidatorLoader(validatorSources, config.getGraffitiProvider());
   }
 
+  private static void addMutableValidatorSource(
+      final Spec spec,
+      final ValidatorConfig config,
+      final SlashingProtector slashingProtector,
+      final AsyncRunner asyncRunner,
+      final List<ValidatorSource> validatorSources,
+      final Optional<DataDirLayout> validatorPath) {
+    DataDirLayout validatorDir = validatorPath.get();
+    Path keystorePath = ValidatorClientService.getKeystoreValidatorPath(validatorDir);
```

Comment:
this is why that function needs to be named well. because it's hard to tell here what we're calling a path that can update.

If the validator path is an optional, we want to guard against it being empty.

also this is the dataDir layout, not an optional of the validator path, it could probably be clearer name as something like maybeDataDir.